pydrive2/auth.py

```diff
@@ -571,22 +571,27 @@ def LoadClientConfigFile(self, client_config_file=None):
             pass  # The service auth fields are not present, handling code can go here.
 
     def LoadServiceConfigSettings(self):
-        """Loads client configuration from settings file.
+        """Loads client configuration from settings.
         :raises: InvalidConfigError
         """
-        for file_format in ["json", "pkcs12"]:
-            config = f"client_{file_format}_file_path"
+        configs = [
+            "client_json_file_path",
+            "client_json_dict",
+            "client_json",
+            "client_pkcs12_file_path",
+        ]
+
+        for config in configs:
             value = self.settings["service_config"].get(config)
             if value:
                 self.client_config[config] = value
                 break
```

Comment:
We can probably transfer all of the options instead of trying to pick the first best one. But keeping similar to how it was before for now.